docs/luigi-core-api.md

```diff
@@ -0,0 +1,192 @@
+# Luigi Core API
+
+## Luigi.config()
+
+<!-- Generated by documentation.js. Update this documentation by updating the source code. -->
+
+### Configuration
+
+#### setConfig
+
+Sets the configuration for Luigi initially. Can also be called at a later point in time again to update the configuration.
+
+##### Parameters
+
+-   `configInput` **[Object](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/Object)?** the Luigi Core configuration object
+
+##### Examples
+
+```javascript
+Luigi.config().setConfig({
+  navigation: {
+    nodes: () => [
+      {
+        pathSegment: 'home',
+        label: 'Home',
+        children: [
+          {
+            pathSegment: 'hello',
+            label: 'Hello Luigi!',
+            viewUrl: '/assets/basicexternal.html'
+          }
+        ]
+      }
+    ]
+  },
+  routing: {
+    useHashRouting: true
+  }
+})
+```
+
+#### getConfigValue
+
+Gets value of the given property on Luigi config object. Target can be a value or a synchronous function.
+
+##### Parameters
+
+-   `property` **[string](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/String)?** the object traversal path
+
+##### Examples
+
+```javascript
+Luigi.config().getConfigValue('auth.use')
```

Comment:
Luigi.config() is not made available globally; this call leads to "TypeError: Luigi.config is not a function". The config related functions are available on root level of Luigi object.